planner/core/task.go

```diff
@@ -1458,7 +1458,12 @@ func CheckAggCanPushCop(sctx sessionctx.Context, aggFuncs []*aggregation.AggFunc
 	client := sctx.GetClient()
 	ret := true
 	reason := ""
-	for _, aggFunc := range aggFuncs {
+	for i := range aggFuncs {
+		aggFunc := aggFuncs[i]
+		if storeType == kv.TiFlash {
+			aggFunc = aggFunc.Clone()
+			aggFunc.WrapCastForAggArgs(sctx)
```

Comment:
Should double check if `aggFunc.WrapCastForAggArgs(sctx)` will introduce side effects on the original aggFunc, since `aggFunc.Clone()` is not 100% deep copy, for example, the `RetType` of expression is just shallow copy.